### PR TITLE
create resource aci_bulk_epg_to_static_path

### PR DESCRIPTION
**Pull request by xinyuezhao**
- create resource aci_bulk_epg_to_static_path which allow multiple static path to be created in one api call
- create example tf plan for this resource

### CODE REVIEW

File: aci/resource_aci_fvrsbulkpathatt.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+		newStaticPathSet = append(newStaticPathSet, newStaticPathMap)
+	}
+	d.Set("static_path", newStaticPathSet)
+	return d
+}
+
+func bulkStaticPathRequest(aciClient *client.Client, method string, path string, body *container.Container) (*container.Container, diag.Diagnostics) {
+	url := "/api/mo/" + path + ".json"
+	if method == "GET" {
+		url += "?rsp-subtree=children"
```
[the remark this answers is not in the document]
Comment (lhercot) [reply aci/resource_aci_fvrsbulkpathatt.go:215]: https://{{APIC1_PUB_IP}}/api/mo/uni/tn-sabari-test/ap-sabari-app-test/epg-test-epg.json?rsp-subtree=children&rsp-subtree-class=fvRsPathAtt
https://{{APIC1_PUB_IP}}/api/class/uni/tn-sabari-test/ap-sabari-app-test/epg-test-epg/fvRsPathAtt.json
https://{{APIC1_PUB_IP}}/api/mo/uni/tn-sabari-test/ap-sabari-app-test/epg-test-epg.json?query-target=children&target-subtree-class=fvRsPathAtt

File: aci/resource_aci_fvrsbulkpathatt.go [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+			staticPathContent["mode"] = mode
+		}
+		if primaryEncap, ok := staticPath["primary_encap"]; ok {
+			staticPathContent["primaryEncap"] = primaryEncap
+		}
+		if status == "delete" {
+			staticPathContent["status"] = "deleted"
+		}
+		//TODO: make map[string]string
+		log.Printf("[DEBUG]: primaryEncap in staticPathPayload %s", staticPath["primary_encap"])
```
Comment (lhercot) [inline aci/resource_aci_fvrsbulkpathatt.go:294]: not needed

**Review by lhercot — CHANGES_REQUESTED** [review]

File: aci/resource_aci_fvrsbulkpathatt.go [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+	}
+}
+
+func resourceAciBulkStaticPathCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] StaticPath: Beginning Creation")
+
+	aciClient := m.(*client.Client)
+
+	ApplicationEPGDn := d.Get("application_epg_dn").(string)
+	d.Get("static_path").(*schema.Set).List()
```
[the remark this answers is not in the document]
Comment (xinyuezhao) [reply aci/resource_aci_fvrsbulkpathatt.go:93]: removed

File: aci/resource_aci_fvrsbulkpathatt.go [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+	if diags.HasError() {
+		return nil, diags
+	}
+	//TODO: get staticPathSet in one search func, Search("imdata", className, "children").Index(0).Data().([]interface{})
+	staticPathSet := resp.S("imdata").Index(0).S(className, "children").S("fvRsPathAtt", "attributes")
+	if staticPathSet == nil {
+		return nil, nil
+	}
+	bulkStaticPath := staticPathSet.Data().([]interface{})
+	return bulkStaticPath, nil
```
[the remark this answers is not in the document]
Comment (xinyuezhao) [reply aci/resource_aci_fvrsbulkpathatt.go:184]: removed

File: aci/resource_aci_fvrsbulkpathatt.go [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+		}
+		newStaticPathMap["encap"] = staticPathMap["encap"].(string)
+		newStaticPathMap["interface_dn"] = staticPathMap["tDn"].(string)
+		newStaticPathSet = append(newStaticPathSet, newStaticPathMap)
+	}
+	d.Set("static_path", newStaticPathSet)
+	return d
+}
+
+func bulkStaticPathRequest(aciClient *client.Client, method string, path string, body *container.Container) (*container.Container, diag.Diagnostics) {
```
[the remark this answers is not in the document]
Comment (xinyuezhao) [reply aci/resource_aci_fvrsbulkpathatt.go:212]: epgDn

File: aci/resource_aci_fvrsbulkpathatt.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (xinyuezhao) [reply aci/resource_aci_fvrsbulkpathatt.go:215]: pick the third one

File: aci/resource_aci_fvrsbulkpathatt.go [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+	_, diags := bulkStaticPathRequest(aciClient, "POST", ApplicationEPGDn, cont)
+	if diags.HasError() {
+		return diags
+	}
+	d.SetId(ApplicationEPGDn)
+	log.Printf("[DEBUG] %s: Update finished successfully", d.Id())
+	return resourceAciBulkStaticPathRead(ctx, d, m)
+}
+
+func getAciBulkStaticPath(aciClient *client.Client, dn string) ([]interface{}, diag.Diagnostics) {
```
Comment (lhercot) [inline aci/resource_aci_fvrsbulkpathatt.go:167]: Is this returning 1 or more static paths? If more than one, rename the function to getAciBulkStaticPaths

File: aci/resource_aci_fvrsbulkpathatt.go [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+				break
+			}
+		}
+		if !found {
+			toDelete = append(toDelete, old)
+		}
+	}
+
+	return toDelete
+}
```
Comment (lhercot) [inline aci/resource_aci_fvrsbulkpathatt.go:246]: not needed. use getOldObjectsNotInNew("interface_dn", oldSet, newSet)

File: aci/resource_aci_fvrsbulkpathatt.go [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+		if mode, ok := staticPath["mode"]; ok {
+			staticPathContent["mode"] = mode
+		}
+		if primaryEncap, ok := staticPath["primary_encap"]; ok {
+			staticPathContent["primaryEncap"] = primaryEncap
+		}
+		if status == "delete" {
+			staticPathContent["status"] = "deleted"
+		}
+		log.Printf("[DEBUG]: primaryEncap in staticPathPayload %s", staticPath["primary_encap"])
```
Comment (lhercot) [inline aci/resource_aci_fvrsbulkpathatt.go:273]: remove

File: aci/resource_aci_fvrsbulkpathatt.go [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+
+	aciClient := m.(*client.Client)
+	ApplicationEPGDn := d.Get("application_epg_dn").(string)
+
+	dn := d.Id()
+	bulkStaticPath, diags := getAciBulkStaticPath(aciClient, dn)
+	if diags.HasError() {
+		return diags
+	}
+	d = setAciBulkStaticPath(bulkStaticPath, d)
```
Comment (lhercot) [inline aci/resource_aci_fvrsbulkpathatt.go:292]: don't think this is needed.

File: examples/aci_bulk_epg_to_static_path/main.tf [hunk 10]
```diff
@@ -0,0 +1,65 @@
+terraform {
```
Comment (lhercot) [inline examples/aci_bulk_epg_to_static_path/main.tf:1]: run terraform fmt to reformat it.

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**